### DISCUSSION TITLE
Detail filter problem

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5756] [Help]
![Filtro_Panta V2](https://user-images.githubusercontent.com/42218266/120054973-37e22580-c001-11eb-9dfd-217c871f18e2.png)
Hello community

I have implemented the filter in Master / Details, according to the solution proposed by "brunobola" .... But apparently something is missing because I can't get the filter to work. Here I leave the programmed, 

![Filtro_Panta](https://user-images.githubusercontent.com/42218266/120052647-45dd7980-bff4-11eb-9334-4f2b7fad981c.png)

XYZEditor.ts
-----------
/// <reference path="../../Common/Helpers/GridEditorBase.ts" />

namespace SegurosMAPC.BaseDatos {

    import fld = ImportarCargaResultadoRow.Fields; 

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.filterable()
    export class ImportarCargaResultadoEditor extends Common.GridEditorBase<ImportarCargaResultadoRow> {
        protected getColumnsKey() { return "BaseDatos.ImportarCargaResultado"; }
        protected getDialogType() { return ImportarCargaResultadoDialog; }
        protected getLocalTextPrefix() { return ImportarCargaResultadoRow.localTextPrefix; }

        constructor(container: JQuery) {
            super(container);
            this.slickGrid.onSort.subscribe(function (e, args) {
                SegurosMAPC.GridHelper.sortGridFunction((args.grid as Slick.Grid), args.sortCols[0], args.sortCols[0].sortCol.field);

                //(args.grid as Slick.Grid).init();
                (args.grid as Slick.Grid).invalidateAllRows();
                (args.grid as Slick.Grid).invalidate();
                (args.grid as Slick.Grid).render();
                (args.grid as Slick.Grid).resizeCanvas();
            });
        }

        protected validateEntity(row: ImportarCargaResultadoRow, id: number) {
            if (!super.validateEntity(row, id))
                return false;
            row.EncNumero = Q.toId(row.EncNumero);

            return true;
        }

        protected getAddButtonCaption() {
            return "Validacion";
        }

        protected getButtons(): Serenity.ToolButton[] {
            var buttons = super.getButtons();
            buttons.splice(Q.indexOf(buttons, x => x.cssClass == "add-button"), 1);
            return buttons;
        }

       
        //
        protected enableFiltering() {
            return true;
        }

        protected layout() {
            super.layout();

            var sortCols = this.slickGrid.getSortColumns();

            SegurosMAPC.GridHelper.sortGridFunction(this.slickGrid, sortCols[0], sortCols[0].columnId);
        }

        protected getDefaultSortBy() {
            //Columns to apply first sort on layout
            return ["ValNumero"];
        }

         protected onViewSubmit() {
            this.view.beginUpdate();
            let filter = function filter(item, args) {
                var linha = (item as ImportarCargaResultadoRow);                            
                if (item["Numero Linea"] == fld.ValNumLinea)
                    return true;  
                if (item["Cobertura"] == fld.ValCobertura)
                    return true;
                return false;
            }
            this.view.setFilter(filter);
            this.view.endUpdate();
            return true;
        }
       
    }
}

GridHelper (Function)
----------
namespace SegurosMAPC.GridHelper {

    export function sortGridFunction(grid: Slick.Grid, column: any, field: any) {
        grid.getData().sort(function (a, b) {
            var result = a[field] > b[field] ? 1 :
                a[field] < b[field] ? -1 :
                    0;
            return column.sortAsc ? result : -result;
        });
    }
}

XYZRow
---------
 [ConnectionKey("BaseDatos"), Module("BaseDatos"), TableName("[dbo].[vw_web_carga_manual_resultado]")]
    [DisplayName("Resultado de Carga"), InstanceName("Resultado de Carga")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    public sealed class ImportarCargaResultadoRow : Row, IIdRow, INameRow
    {
.....
.....
  [DisplayName("Numero Linea"), Column("val_num_linea"), QuickFilter]
        public Int32? ValNumLinea
        {
            get { return Fields.ValNumLinea[this]; }
            set { Fields.ValNumLinea[this] = value; }
        }
[DisplayName("Cobertura"), Column("val_cobertura"), Size(2), NotNull, QuickFilter]
        public String ValCobertura
        {
            get { return Fields.ValCobertura[this]; }
            set { Fields.ValCobertura[this] = value; }
        }
....
....
.....

if there is someone who knows the solution it would be of great help.
Atte.-

**Comment by eissa1005** [comment]
@patriciomh1965 
 You are missing this 
protected getQuickFilters() {
            var flt = super.getQuickFilters();

            var q = Q.parseQueryString();
            if (q["cat"]) {
                var category = Q.tryFirst(flt, x => x.field == "CategoryID");
                category.init = e => {
                    e.element.getWidget(Serenity.LookupEditor).value = q["cat"];
                };
            }

            return flt;
        }

**Reply by patriciomh1965** [reply]
Hello, I implemented it but I am still with the filter problem. I keep looking for the solution.
Thanks for your help.

**Reply by axelcasauran** [reply]
hello.. have you solve this issue? i am facing the same problem

**Comment by patriciomh1965** [comment]
Hello 

I couldn't solve the problem, unfortunately
